conan/tools/env/environment.py

```diff
@@ -414,8 +414,11 @@ def save_bat(self, file_location, generate_deactivate=True):
             {deactivate}
             """).format(deactivate=deactivate if generate_deactivate else "")
         result = [capture]
+        location = os.path.dirname(file_location)
         for varname, varvalues in self._values.items():
             value = varvalues.get_str("%{name}%", subsystem=self._subsystem, pathsep=self._pathsep)
+            # To make the script relocatable
+            value = value.replace(location, "%~dp0")
```

Comment:
This was easy to make Env generators to be relocatable for Windows/.bat, but it is challenging to do for .sh in Linux, due to not existing a generic pure shell (no bash only) solution